### PR TITLE
Add EqualsIgnoreCase string builtin function

### PR DESCRIPTION
**Pull request by ranger-turtle**
I have added EqualsIgnoreCase string builtin function which would be faster way for case insensitive string comparison than the workaround I have mentioned in my comment in #508 .

### CODE REVIEW

**Comment by binarycow** [comment]
(Note - I'm not the maintainer of Scriban - just another occasional contributer) 

You beat me to it!

While you're at it, I have a suggestion, if you have the time...

Add an overload for the normal equals method that accepts `StringComparison`.

Could also do that for all of the other methods, where the .NET method also accepts `StringComparison`. 

IIRC, Scriban supports optional parameters, so it would be as easy as copy/pasting the `StringComparison` parameter in three places for each method:

- Parameter list
- Method call
- XML docs  

If Scriban doesn't support optional parameters, then.... Maybe I'll add that?

**Comment by ranger-turtle** [comment]
I considered making general `equals` builtin function with `StringComparison` type parameter but I decided to make the `EqualsIgnoreCase` function instead, since case sensitive comparison can be accomplished using `==` operator and I found making `equals` unnecessary.

However, I think I may add an optional parameter to such functions, like `StartsWith` or `Contains`. I will wait for @xoofx 's answer to ensure if this PR makes sense according to him.

File: src/Scriban/Functions/StringFunctions.cs [hunk 1]
```diff
@@ -305,6 +305,25 @@ public static bool EndsWith(string text, string value)
...
 
+        /// <summary>
+        /// Returns a boolean indicating whether the input string is equal to specified string 'value'. Comparison is case insensitive.
+        /// </summary>
+        /// <param name="text">The input string</param>
+        /// <param name="value">The string to compare</param>
+        /// <returns><c>true</c> if `text` is equal to string `value`, ignoring case</returns>
+        /// <remarks>
+        /// ```scriban-html
+        /// {{ "Scriban" | string.ends_with "SCRIBAN" }}
```
Comment (xoofx) [inline src/Scriban/Functions/StringFunctions.cs:316]:
```suggestion
        /// {{ "Scriban" | string.equals_ignore_case "SCRIBAN" }}
```

**Comment by xoofx** [comment]
Adding `string.equals_ignore_case` is fine. Tests are failing (but due to the flood of logs, can't see it easily in GitHub, you need to check this locally)

I would prefer not to change existing methods. For 6+ years nobody has been looking for configurable string comparisons, so this is not necessary to change all these. Users can always create custom methods on their side.

**Pull request merged**